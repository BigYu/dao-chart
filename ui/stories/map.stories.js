import _ from 'lodash';
import React from 'react';
import { storiesOf } from '../.storybook/facade';
import Map from '../src/component/map';

function randomData() {
  return Math.round(Math.random() * 1000);
}

const locs = [
  '北京',
  '天津',
  '上海',
  '重庆',
  '河北',
  '河南',
  '云南',
  '辽宁',
  '黑龙江',
  '湖南',
  '安徽',
  '山东',
  '新疆',
  '江苏',
  '浙江',
  '江西',
  '湖北',
  '广西',
  '甘肃',
  '山西',
  '内蒙古',
  '陕西',
  '吉林',
  '福建',
  '贵州',
  '广东',
  '青海',
  '西藏',
  '四川',
  '宁夏',
  '海南',
  '台湾',
  '香港',
  '澳门',
];

const source = _.map(locs, loc => ([
  loc, randomData(), randomData(),
]));

source.unshift(['Location', 'iPhone3', 'iPhone4']);

storiesOf('Map Chart', module)
  .add('Simple map', () => (
    <Map source={source} style={{ height: '800px', width: '100%' }} />
  ));
