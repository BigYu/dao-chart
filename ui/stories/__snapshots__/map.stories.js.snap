// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Simple map 1`] = `
<EchartsReact
  className=""
  echarts={Object {}}
  lazyUpdate={false}
  loadingOption={null}
  notMerge={false}
  onChartReady={[Function]}
  onEvents={Object {}}
  option={
    Object {
      "legend": Object {},
      "series": Array [
        Object {
          "data": Array [
            Object {
              "name": "北京",
              "value": 272,
            },
            Object {
              "name": "天津",
              "value": 508,
            },
            Object {
              "name": "上海",
              "value": 297,
            },
            Object {
              "name": "重庆",
              "value": 913,
            },
            Object {
              "name": "河北",
              "value": 850,
            },
            Object {
              "name": "河南",
              "value": 934,
            },
            Object {
              "name": "云南",
              "value": 620,
            },
            Object {
              "name": "辽宁",
              "value": 975,
            },
            Object {
              "name": "黑龙江",
              "value": 532,
            },
            Object {
              "name": "湖南",
              "value": 839,
            },
            Object {
              "name": "安徽",
              "value": 827,
            },
            Object {
              "name": "山东",
              "value": 297,
            },
            Object {
              "name": "新疆",
              "value": 994,
            },
            Object {
              "name": "江苏",
              "value": 951,
            },
            Object {
              "name": "浙江",
              "value": 307,
            },
            Object {
              "name": "江西",
              "value": 657,
            },
            Object {
              "name": "湖北",
              "value": 844,
            },
            Object {
              "name": "广西",
              "value": 213,
            },
            Object {
              "name": "甘肃",
              "value": 841,
            },
            Object {
              "name": "山西",
              "value": 513,
            },
            Object {
              "name": "内蒙古",
              "value": 636,
            },
            Object {
              "name": "陕西",
              "value": 546,
            },
            Object {
              "name": "吉林",
              "value": 957,
            },
            Object {
              "name": "福建",
              "value": 441,
            },
            Object {
              "name": "贵州",
              "value": 306,
            },
            Object {
              "name": "广东",
              "value": 892,
            },
            Object {
              "name": "青海",
              "value": 363,
            },
            Object {
              "name": "西藏",
              "value": 876,
            },
            Object {
              "name": "四川",
              "value": 343,
            },
            Object {
              "name": "宁夏",
              "value": 7,
            },
            Object {
              "name": "海南",
              "value": 788,
            },
            Object {
              "name": "台湾",
              "value": 452,
            },
            Object {
              "name": "香港",
              "value": 776,
            },
            Object {
              "name": "澳门",
              "value": 17,
            },
          ],
          "label": Object {
            "emphasis": Object {
              "show": true,
            },
            "normal": Object {
              "show": true,
            },
          },
          "mapType": "china",
          "name": "iPhone3",
          "roam": false,
          "type": "map",
        },
        Object {
          "data": Array [
            Object {
              "name": "北京",
              "value": 973,
            },
            Object {
              "name": "天津",
              "value": 129,
            },
            Object {
              "name": "上海",
              "value": 23,
            },
            Object {
              "name": "重庆",
              "value": 488,
            },
            Object {
              "name": "河北",
              "value": 334,
            },
            Object {
              "name": "河南",
              "value": 361,
            },
            Object {
              "name": "云南",
              "value": 172,
            },
            Object {
              "name": "辽宁",
              "value": 683,
            },
            Object {
              "name": "黑龙江",
              "value": 567,
            },
            Object {
              "name": "湖南",
              "value": 435,
            },
            Object {
              "name": "安徽",
              "value": 881,
            },
            Object {
              "name": "山东",
              "value": 85,
            },
            Object {
              "name": "新疆",
              "value": 914,
            },
            Object {
              "name": "江苏",
              "value": 667,
            },
            Object {
              "name": "浙江",
              "value": 946,
            },
            Object {
              "name": "江西",
              "value": 160,
            },
            Object {
              "name": "湖北",
              "value": 398,
            },
            Object {
              "name": "广西",
              "value": 825,
            },
            Object {
              "name": "甘肃",
              "value": 512,
            },
            Object {
              "name": "山西",
              "value": 221,
            },
            Object {
              "name": "内蒙古",
              "value": 570,
            },
            Object {
              "name": "陕西",
              "value": 651,
            },
            Object {
              "name": "吉林",
              "value": 104,
            },
            Object {
              "name": "福建",
              "value": 394,
            },
            Object {
              "name": "贵州",
              "value": 606,
            },
            Object {
              "name": "广东",
              "value": 1,
            },
            Object {
              "name": "青海",
              "value": 346,
            },
            Object {
              "name": "西藏",
              "value": 458,
            },
            Object {
              "name": "四川",
              "value": 917,
            },
            Object {
              "name": "宁夏",
              "value": 446,
            },
            Object {
              "name": "海南",
              "value": 305,
            },
            Object {
              "name": "台湾",
              "value": 710,
            },
            Object {
              "name": "香港",
              "value": 754,
            },
            Object {
              "name": "澳门",
              "value": 732,
            },
          ],
          "label": Object {
            "emphasis": Object {
              "show": true,
            },
            "normal": Object {
              "show": true,
            },
          },
          "mapType": "china",
          "name": "iPhone4",
          "roam": false,
          "type": "map",
        },
      ],
      "tooltip": Object {
        "trigger": "item",
      },
      "visualMap": Object {
        "calculable": true,
        "left": "left",
        "max": 1909,
        "min": 0,
        "text": Array [
          "高",
          "低",
        ],
        "top": "bottom",
      },
    }
  }
  opts={Object {}}
  showLoading={false}
  source={
    Array [
      Array [
        "Location",
        "iPhone3",
        "iPhone4",
      ],
      Array [
        "北京",
        272,
        973,
      ],
      Array [
        "天津",
        508,
        129,
      ],
      Array [
        "上海",
        297,
        23,
      ],
      Array [
        "重庆",
        913,
        488,
      ],
      Array [
        "河北",
        850,
        334,
      ],
      Array [
        "河南",
        934,
        361,
      ],
      Array [
        "云南",
        620,
        172,
      ],
      Array [
        "辽宁",
        975,
        683,
      ],
      Array [
        "黑龙江",
        532,
        567,
      ],
      Array [
        "湖南",
        839,
        435,
      ],
      Array [
        "安徽",
        827,
        881,
      ],
      Array [
        "山东",
        297,
        85,
      ],
      Array [
        "新疆",
        994,
        914,
      ],
      Array [
        "江苏",
        951,
        667,
      ],
      Array [
        "浙江",
        307,
        946,
      ],
      Array [
        "江西",
        657,
        160,
      ],
      Array [
        "湖北",
        844,
        398,
      ],
      Array [
        "广西",
        213,
        825,
      ],
      Array [
        "甘肃",
        841,
        512,
      ],
      Array [
        "山西",
        513,
        221,
      ],
      Array [
        "内蒙古",
        636,
        570,
      ],
      Array [
        "陕西",
        546,
        651,
      ],
      Array [
        "吉林",
        957,
        104,
      ],
      Array [
        "福建",
        441,
        394,
      ],
      Array [
        "贵州",
        306,
        606,
      ],
      Array [
        "广东",
        892,
        1,
      ],
      Array [
        "青海",
        363,
        346,
      ],
      Array [
        "西藏",
        876,
        458,
      ],
      Array [
        "四川",
        343,
        917,
      ],
      Array [
        "宁夏",
        7,
        446,
      ],
      Array [
        "海南",
        788,
        305,
      ],
      Array [
        "台湾",
        452,
        710,
      ],
      Array [
        "香港",
        776,
        754,
      ],
      Array [
        "澳门",
        17,
        732,
      ],
    ]
  }
  style={
    Object {
      "height": "800px",
      "width": "100%",
    }
  }
  theme={null}
/>
`;
